src/main/java/Duke.java

```diff
@@ -1,10 +1,126 @@
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The Duke class contains the main method.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    /**
+     * The constructor for the Duke class.
+     */
+    public Duke() {
+        ui = new Ui();
+        tasks = new TaskList();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    /**
+     * The main method for Duke.
+     * @param args For the main method.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public static void main(String[] args) throws DukeException, IOException {
+        new Duke().run();
+    }
+
+
+    /**
+     * To run the duke program.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public void run() throws DukeException, IOException {
+        storage.loadData();
+        ui.welcomeMessage();
+        ui.reply("    What can I do for you");
+
+        Scanner sc = new Scanner(System.in);
+        String command;
+        while (!(command = sc.nextLine()).equals("bye")) {
```

Comment:
This is a rather unconventional statement.
I would recommend breaking it up into multiple parts

On the line above, you could add the statement
command = sc.nextLine()

then you could adjust this line to:
while (!command.equals("bye") {

and finally, you could add the statement at the end of the while loop:
command = sc.nextLine()